src/block-components/image/edit.js

```diff
@@ -185,7 +204,15 @@ const Controls = props => {
 						//TODO: Add a working video
 						title: __( 'Image height', i18n ),
 						description: __( 'Adjusts the image height', i18n ),
-					} }s
+					} }
+					// Set the width to auto if the aspect ratio is set and the width is currently empty
+					// to allow the aspect ratio to take effect with height changes.
+					onChange={ value => {
+						setAttributes( { imageHeight: value } )
+						if ( attributes.imageAspectRatio !== '' && attributes.imageWidth === '' ) {
+							setAttributes( { imageWidth: 'auto' } )
```

Comment:
We want to call `setAttributes` only **once** when a value changes. In the block editor, whenever `setAttributes` is called, it creates a new entry in the **changes history**, since it's called twice here, you'll have to undo two times.